### PR TITLE
Template interpreter

### PR DESCRIPTION
**Pull request by Spayralbe**
New code includes the template_interpreter function and one small bugfix for ColumnMapping.py
@jochemb please have a look

### CODE REVIEW

File: tmtk/toolbox/template_interpreter.py [hunk 1]
```diff
@@ -0,0 +1,719 @@
+# coding: utf-8
```
Comment (OlafMeuwese) [inline tmtk/toolbox/template_interpreter.py]: you could consider splitting the file into different files for the classes.  
700 lines of code containing multiple classes could be a suboptimal reading experience

Comment (Spayralbe) [inline tmtk/toolbox/template_interpreter.py]: Thanks! I will consider it.

Comment (Spayralbe) [inline tmtk/toolbox/template_interpreter.py]: @OlafMeuwese Code has been split.

**Comment by jochemb** [comment]
@Spayralbe I'd like to see some tests with the test template that was used to develop this.

File: tmtk/utils/filebase.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import os
 import pandas as pd
+from IPython.display import display, HTML
```
Comment (jochemb) [inline tmtk/utils/filebase.py]: Why import HTML here if you do not use it?

File: tests/interpret_templates_tests.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    def tearDown(self):
+        pass
+
+    def test_study_load(self):
+        assert self.study.params_path
+
+    def test_data_volumes(self):
+        assert self.study.study_id == self.control_study.study_id
+
+    def test_file_equality(self):
```
Comment (Spayralbe) [inline tests/interpret_templates_tests.py]: STD will now be left empty and word mapping sorted on the first 3 cols

File: tmtk/toolbox/interpret_templates.py [hunk 4]
```diff
@@ -0,0 +1,592 @@
...
+
+def write_hd_data_params(experiment):
+    """Write HD data params file."""
+    params_output_path = os.path.join(experiment.output_dir, experiment.hd_data_params_name)
+
+    with open(params_output_path, "w") as hd_data_params_file:
+        if experiment.hd_type:
+            hd_data_params_file.write("DATA_FILE=" + experiment.hd_type + "_data.tsv" + "\n")
+        hd_data_params_file.write("DATA_TYPE=" + "R" + "\n")
+        # hd_data_params_file.write("LOG_BASE="+"2"+"\n")
```
Comment (Spayralbe) [inline tmtk/toolbox/interpret_templates.py]: Yes, will currently not be used, but might be implemented later.

File: tmtk/toolbox/template_data.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        self.sheets = HighDim.Sheets()
+
+    def read_hd_file_template(self, source_dir, hd_template):
+        """Try to read the specified template file and send to sheet loading method."""
+        template_path = os.path.join(source_dir, hd_template)
+        try:
+            hd_template_workbook = pd.ExcelFile(template_path, comment="#", dtype=object)
+        except FileNotFoundError:
+            raise Validity.TemplateException("Could not find high-dim template file at: {0}".format(template_path))
+        # except XLRDError:
```
Comment (Spayralbe) [inline tmtk/toolbox/template_data.py]: Preferably do catch it, but it is not compatible with all python3 versions.

**Pull request merged**